decompressed/base/etc/init.d/rootdevice

```diff
@@ -252,56 +203,97 @@ install_gui() {
 
-  if [ "$bootedbank" == "bank_2" ]; then
-    if [ -f $env_config_bank1 ]; then
-      get_config_bank1=$(< $env_config_bank1 grep bank_check | awk '{print $3}' | sed "s/'//g")
-      if [ -n "$get_config_bank1" ] && [ ! "$(uci -q get modgui.var.bank_check)" ]; then
-        uci set modgui.var.bank_check="$get_config_bank1"
+  if [[ "$bootedbank" != "$activebank" && "$bootedbank" == "bank_2" ]]; then
+    #TODO: add checks to be sure the bank_1 is unbootable (is this really needed?)
+    #TODO: add checks to be sure the bank_1 overlay is SAFE and bootable (empty/prerooted?)
+    echo "  OBP sanity check succeeded continue boot complete..." >/dev/console
+  else
+    #OBP not OK, check if was an user choice otherwise automatically apply (this should be the case called also on first install)
+
+    if [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+      #we must inizialize the check_obp var as is not found
+      if [ -f $notbooted_modgui_configfile ]; then
+        #if we are in a clean bank_2 overlay but in the notbooted bank overlay there was a modgui config file restore var from it
+        get_config_notbooted=$(< $notbooted_modgui_configfile grep check_obp | awk '{print $3}' | sed "s/'//g")
+        if [ -n "$get_config_notbooted" ] && [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+          uci set modgui.var.check_obp="$get_config_notbooted"
+          uci commit modgui
+        fi
+      else
+        #inizialize check_obp default value and only if cannot read the var from elsewhere (inactive bank config)
+        uci set modgui.var.check_obp="1"
         uci commit modgui
       fi
-    else
-      uci set modgui.var.bank_check="1"
-      uci commit modgui
     fi
-    if [ "$(uci -q get modgui.var.bank_check)" == "0" ]; then
-      logger_command "Firmware in Bank_2 but check_bank set to 0... Not transferring"
-      echo "  Firmware in Bank_2 but check_bank set to 0... Not transferring" >/dev/console
+
+    if [ "$(uci -q get modgui.var.check_obp)" == "0" ]; then
+      logger_command "OBP sanity check FAILED but check_obp set to 0... Not acting"
+      echo "  OBP sanity check FAILED but check_obp set to 0... Not acting" >/dev/console
     else
-      logger_command "Firmware in Bank_2... Transferring"
-      echo "  Firmware in Bank_2... Transferring" >/dev/console
-      transfer_bank
+      logger_command "OBP sanity check FAILED start autoplanning..."
+      echo "  OBP sanity check FAILED start autoplanning..." >/dev/console
+
+      if [ "$bootedbank" == "bank_2" ]; then
+        # Activate bank_1
+        echo bank_1 > /proc/banktable/active
+        # Erase firmware in bank_1
+        mtd erase bank_1
+      else
+        # clone bank_1 into bank_2
+        mtd write bank_1 bank_2
+        md5_bank_1=$(md5sum "/dev/mtd3" | awk '{ print $1 }')
+        md5_bank_2=$(md5sum "/dev/mtd4" | awk '{ print $1 }')
+        if [ "$md5_bank_1" == "$md5_bank_2" ]; then
```

Comment:
@LuKePicci credi basti in questo modo? hai idee migliori sul cosa fare in caso di fallimento? gli mtd li lascio hardcoddati?

Reply:
c'era una funzione di tch che faceva la traduzione bank_2->mtd4 e simili nel file che montava l'overlay. In quel caso prendeva in input userfs rootfs_data e restituiva il primo mtd corrispondente, puoi riciclare quella.

mtd write di sicuro non prende bank_1 come input, mi pare che write voglia un file come argomento

In caso di fallimento ti conviene fermare tutto. Ritentare è pericoloso, rischi di finire in un loop interminabile se non gestisci laboriosamente un numero di tentativi limitato